tests/modules/redshift_datasets/conftest.py

```diff
@@ -0,0 +1,201 @@
+import os
+import pytest
+
+from dataall.base.context import set_context, dispose_context, RequestContext
+
+from dataall.modules.redshift_datasets.services.redshift_connection_service import RedshiftConnectionService
+from dataall.modules.redshift_datasets.services.redshift_dataset_service import RedshiftDatasetService
+
+ENVNAME = os.environ.get('envname', 'pytest')
+
+
+class MockRedshiftDataClient:
+    def get_redshift_connection_database(self, *args, **kwargs):
+        return True
+
+    def list_redshift_schemas(self, *args, **kwargs):
+        return ['public', 'dev']
+
+    def list_redshift_tables(self, *args, **kwargs):
+        return [
+            {'name': 'table1', 'type': 'TABLE'},
+            {'name': 'table2', 'type': 'TABLE'},
+            {'name': 'table3', 'type': 'TABLE'},
+            {'name': 'table4', 'type': 'TABLE'},
+        ]
+
+    def list_redshift_table_columns(self, *args, **kwargs):
+        return [
+            {'name': 'column1', 'type': 'VARCHAR', 'nullable': True},
+            {'name': 'column2', 'type': 'INTEGER', 'nullable': False},
+            {'name': 'column3', 'type': 'DOUBLE', 'nullable': True},
+            {'name': 'column4', 'type': 'BOOLEAN', 'nullable': False},
+        ]
+
+
+class MockRedshiftClient:
+    def describe_cluster(self, *args, **kwargs):
+        return {'ClusterIdentifier': 'cluster_id_1', 'ClusterStatus': 'available'}
+
+
+class MockRedshiftServerlessClient:
+    def get_namespace_by_id(self, *args, **kwargs):
+        return {'namespaceId': 'XXXXXXXXXXXXXX', 'namespaceName': 'namespace_name_1'}
+
+    def list_workgroups_in_namespace(self, *args, **kwargs):
+        return [
+            {
+                'workgroupName': 'workgroup_name_1',
+                'workgroupArn': 'arn:aws:redshift-serverless:eu-west-1:XXXXXXXXXXXXXX:workgroup/workgroup_name_1',
+            }
+        ]
+
+    def get_workgroup_arn(self, *args, **kwargs):
+        return 'arn:aws:redshift-serverless:eu-west-1:XXXXXXXXXXXXXX:workgroup/workgroup_name_1'
+
+
+@pytest.fixture(scope='function')
+def patch_redshift(module_mocker):
```

Comment:
you create this fixture per function but you use `module_mocker` instead of `mocker`, does that mean that the objects will remain patched for the entire module? is that what you want to achieve?

Reply:
[[docs](https://pytest-mock.readthedocs.io/en/latest/usage.html)], It is also possible to use mocking functionality from fixtures of other scopes using the appropriate fixture (module_mocker, class_mocker...) What leads me to think that we can use module_mocker from the function scope, but as you pointed out it can be already pathing something that we are not aware is being patched